src/main/java/org/jenkinsci/plugins/oic/OicSecurityRealm.java

```diff
@@ -1070,6 +1096,32 @@ private UsernamePasswordAuthenticationToken loginAndSetUserData(
             user.setFullName(fullName);
         }
 
+        // Set avatar if possible
+        if (StringUtils.isNotBlank(avatarFieldName)) {
+            try {
+
+                // Extract avatar from the userInfo
+                String avatarUrl = determineStringField(avatarFieldExpr, idToken, userInfo);
```

Comment:
I'm not sure this should be configurable.  The configuration is already overwhelmingly complex and the reason for overriding the other things is that there is no standard for them (or there is but there are good reasons to use something else - e.g. `sub` is not a very useable username so email is generally used).